packages/pay/src/index.ts

```diff
@@ -217,25 +221,38 @@ export const setupPayment = async (options: SetupOptions): Promise<ResolvedPayme
 export const startQuote = async (options: QuoteOptions): Promise<Quote> => {
   const rateProbe = new RateProbe(options)
   const { log } = rateProbe
-  const { invoice, destinationAsset } = options.destination
+  const { incomingPayment, destinationAsset } = options.destination
 
   // Validate the amounts to set the target for the payment
   let target: {
     type: PaymentType
     amount: PositiveInt
   }
 
-  if (invoice) {
-    const remainingToDeliver = Int.from(invoice.amountToDeliver - invoice.amountDelivered)
+  if (incomingPayment) {
+    if (incomingPayment.state === IncomingPaymentState.Completed) {
+      log.debug('quote failed: incoming payment was completed.')
+      // In incoming payment case, STREAM connection is yet to be established since no asset probe
+      throw PaymentError.IncomingPaymentCompleted
+    }
+    if (incomingPayment.state === IncomingPaymentState.Expired) {
+      log.debug('quote failed: incoming payment is expired.')
+      // In incoming payment case, STREAM connection is yet to be established since no asset probe
+      throw PaymentError.IncomingPaymentExpired
+    }
+
+    const remainingToDeliver = Int.from(
+      incomingPayment.amountToDeliver - incomingPayment.amountDelivered
+    )
     if (!remainingToDeliver || !remainingToDeliver.isPositive()) {
-      // Return this error here instead of in `setupPayment` so consumer can access the resolved invoice
+      // Return this error here instead of in `setupPayment` so consumer can access the resolved incoming payment
       log.debug(
-        'quote failed: invoice was already paid. amountToDeliver=%s amountDelivered=%s',
-        invoice.amountToDeliver,
-        invoice.amountDelivered
+        'quote failed: incoming payment was already paid. amountToDeliver=%s amountDelivered=%s',
+        incomingPayment.amountToDeliver,
+        incomingPayment.amountDelivered
       )
-      // In invoice case, STREAM connection is yet to be established since no asset probe
-      throw PaymentError.InvoiceAlreadyPaid
+      // In incoming payment case, STREAM connection is yet to be established since no asset probe
+      throw PaymentError.IncomingPaymentPaid
```

Comment:
:thinking: this _shouldn't_ happen (incoming payment has received full amount but its state isn't `Completed`).
Maybe we just throw `IncomingPaymentCompleted` here too?

Reply:
Yes, I was also contemplating to get rid of this. I'm happy either way: remove it or throw the `IncomingPaymentCompleted` error.